source/multigrid/mg_transfer_matrix_free.cc

```diff
@@ -561,15 +572,29 @@ template <int dim, typename Number>
 void MGTransferBlockMatrixFree<dim,Number>::initialize_constraints
 (const MGConstrainedDoFs &mg_c)
 {
-  matrix_free_transfer.initialize_constraints(mg_c);
+  AssertDimension(matrix_free_transfer_vector.size(),1);
+  matrix_free_transfer_vector[0].initialize_constraints(mg_c);
+  same_for_all = true;
```

Comment:
maybe make this member `const` and have an assert here? Because you can then use incompatible constructor and `initialize_constraints()` pairs.